### PR TITLE
Add ghost locations to the ppx.

### PR DESCRIPTION
**Pull request by Drup**
Requested by @def-lkb to help merlin.

Wait for his approval before merging.

### CODE REVIEW

**Comment by Drup** [comment]
@whitequark : you might want to add it everywhere too.

**Comment by def-lkb** [comment]
I need a few more to check it. Slightly too many expressions are made ghost by the patch in its current state, and merlin is not yet interpreting all locations as it should.

File: ppx/ppx_lwt_ex.ml [hunk 1]
```diff
@@ -337,18 +344,18 @@ let lwt_mapper args =
         else
           default_mapper.expr mapper expr
       | { pexp_desc = Pexp_apply (fn, args); pexp_attributes; pexp_loc } when !log ->
-        default_loc := pexp_loc;
+        default_loc := ghost pexp_loc;
```
Comment (def-lkb) [inline ppx/ppx_lwt_ex.ml]: It's mapping an expression from the input tree, it is a bit too early to set it ghost.

**Comment by Drup** [comment]
Fixed, other issues ?

**Comment by vbmithr** [comment]
@drup @def-lkb Happy with this ? Should we merge it ?

**Comment by Drup** [comment]
It's not ready and it's not important enough to delay the relase.

**Comment by vbmithr** [comment]
@drup: what is the state of this now?

**Comment by Drup** [comment]
I don't know, @def-lkb ?

**Comment by def-lkb** [comment]
Good to me.

**Comment by def-lkb** [comment]
(sorry for previous comment)
Good to me. I updated merlin to handle these cases better (by interpreting ghost_loc differently).

Unless you encountered buggy cases with merlin 2.3.1 when using lwt-ppx, this can be closed.

**Pull request closed without merging**